scolapasta-int-parse/src/radix.rs

```diff
@@ -0,0 +1,252 @@
+use core::num::NonZeroU32;
+
+// Create a lookup table from each byte value (of which the ASCII range is
+// relevant) to the maximum minimum radix the character is valid for.
+#[allow(clippy::cast_possible_truncation)]
+const fn radix_table() -> [u32; 256] {
+    // `u32::MAX` is used as a sentinel such that no valid radix can use this
+    // byte.
+    let mut table = [u32::MAX; 256];
+    let mut idx = 0_usize;
+    loop {
+        if idx >= table.len() {
```

Comment:
Move this check to right before the index increment to prevent overflowing `u8::MAX`

Reply:
done